packages/astro/src/core/add/index.ts

```diff
@@ -0,0 +1,456 @@
+	names = names.map((name) => (CONSTS.ALIASES.has(name) ? CONSTS.ALIASES.get(name)! : name));
+
+	if (configURL) {
+		debug('add', `Found config at ${configURL}`);
+	} else {
+		info(logging, 'add', `Unable to locate a config file, generating one for you.`);
+		configURL = new URL('./astro.config.mjs', projectRoot);
+		await fs.writeFile(fileURLToPath(configURL), CONSTS.CONFIG_STUB, { encoding: 'utf-8' });
+	}
+
+	const integrations = await validateIntegrations(names);
+
+	let ast: t.File | null = null;
+	try {
+		ast = await parseAstroConfig(configURL);
+
+		debug('add', 'Parsed astro config');
+
+		const defineConfig = t.identifier('defineConfig');
+		ensureImport(ast, t.importDeclaration([t.importSpecifier(defineConfig, defineConfig)], t.stringLiteral('astro/config')));
+		wrapDefaultExport(ast, defineConfig);
+
+		debug('add', 'Astro config ensured `defineConfig`');
+
+		for (const integration of integrations) {
+			await addIntegration(ast, integration);
+			debug('add', `Astro config added integration ${integration.id}`);
+		}
+	} catch (err) {
+		debug('add', 'Error parsing/modifying astro config: ', err);
+		return bail(err as Error);
+	}
+
+	let configResult: UpdateResult | undefined;
+	let installResult: UpdateResult | undefined;
+
+	if (ast) {
+		try {
+			configResult = await updateAstroConfig({ configURL, ast, flags, logging });
+		} catch (err) {
+			debug('add', 'Error updating astro config', err);
+			return bail(err as Error);
+		}
+	}
+
+	switch (configResult) {
+		case UpdateResult.cancelled: {
+			info(logging, null, msg.cancelled(`Your configuration has ${bold('NOT')} been updated.`));
+			return;
+		}
+		case UpdateResult.none: {
+			const pkgURL = new URL('./package.json', configURL);
+			if (existsSync(fileURLToPath(pkgURL))) {
+				const { dependencies = {}, devDependencies = {} } = await fs.readFile(fileURLToPath(pkgURL)).then(res => JSON.parse(res.toString()));
+				const deps = Object.keys(Object.assign(dependencies, devDependencies));
+				const missingDeps = integrations.filter(integration => !deps.includes(integration.packageName));
+				if (missingDeps.length === 0) {
+					info(logging, null, msg.success(`Configuration up-to-date.`));
+					return;
+				}
+			}
+
+			info(logging, null, msg.success(`Configuration up-to-date.`));
+			break;
+		}
+	}
+
+	installResult = await tryToInstallIntegrations({ integrations, cwd, flags, logging });
+
+	switch (installResult) {
+		case UpdateResult.updated: {
+			const len = integrations.length;
+			if (integrations.find((integration) => integration.id === 'tailwind')) {
+				const possibleConfigFiles = ['./tailwind.config.cjs', './tailwind.config.mjs', './tailwind.config.js'].map(p => fileURLToPath(new URL(p, configURL)));
+				let alreadyConfigured = false;
+				for (const possibleConfigPath of possibleConfigFiles) {
+					if (existsSync(possibleConfigPath)) {
+						alreadyConfigured = true;
+						break;
+					}
+				}
+				if (!alreadyConfigured) {
```

Comment:
- mentioned in person, but don't overwrite existing config files.
- also came up in person, can you log when this happens so that users aren't surprised?